### PR TITLE
[Issue #1054] Added spec coverage for group_notifier

### PR DESCRIPTION
**Pull request by jm96441n**
added specs for group notifier

# Description 

This PR introduces spec coverage for the `group_notifier` service.

## Corresponding Issue

https://github.com/ifmeorg/ifme/issues/1054

# Screenshots

N/A

# Test Coverage

✅

### CODE REVIEW

File: spec/services/group_notifier_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+describe GroupNotifier do
+  let(:current_user) { double(User, name: 'user', id: 1) }
```
Comment (julianguyen) [inline spec/services/group_notifier_spec.rb]: You can actually use existing FactoryBot values for user and group, but doubles do work too

Comment (jm96441n) [reply spec/services/group_notifier_spec.rb]: that's true, I usually prefer using doubles/mocks but if it's preferred I can use FactoryBot here

File: spec/services/group_notifier_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      uniqueid: "notifier_#{current_user.id}"
+    }.to_json
+  end
+  subject { GroupNotifier.new(group, type, current_user) }
+
+  it 'creates push notifications' do
+    allow(subject).to receive(:create_notification)
+    allow(subject).to receive(:send_email_notification)
+    allow(Notification).to receive_message_chain(:where, :order).and_return(new_notifications)
+    allow(Pusher).to receive(:[]).with("private-#{recipient.id}").and_return(client)
```
Comment (julianguyen) [inline spec/services/group_notifier_spec.rb]: Could this be `Array` instead of `:[]`?

File: spec/services/group_notifier_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    expect(mailer).to receive(:deliver_now)
+  end
+
+  it 'creates notifications' do
+    allow(subject).to receive(:push_notifications)
+    allow(subject).to receive(:send_email_notification)
+    expect(Notification).to receive(:create).with(user_id: recipient.id, uniqueid: "notifier_#{current_user.id}", data: data)
+  end
+
+  after do
```
Comment (julianguyen) [inline spec/services/group_notifier_spec.rb]: I usually put before and after blocks at the top of the file. I've never seen an after at the end, but it makes sense actually.

File: spec/services/group_notifier_spec.rb (hunk 2 above)
Comment (jm96441n) [reply spec/services/group_notifier_spec.rb]: TBH I'm not sure, I'll give it a try

Comment (jm96441n) [reply spec/services/group_notifier_spec.rb]: Actually TIL so ruby access hashes using `.[]('key-here')` as a method:
screenshot from 2018-10-04 20-44-46

Comment (jm96441n) [reply spec/services/group_notifier_spec.rb]: works the same for arrays as well

File: spec/services/group_notifier_spec.rb (hunk 1 above)
Comment (julianguyen) [reply spec/services/group_notifier_spec.rb]: Let's replace things with FactoryBot where it already exists. Thanks!

Comment (jm96441n) [reply spec/services/group_notifier_spec.rb]: Can do! I'll make those changes tonight

**Comment by julianguyen** [comment]
Thanks so much for working on this! Awesome work :) Feel free to merge whenever ready!

**Comment by jm96441n** [comment]
Awesome, thanks for everyone's feedback on this!

**Pull request merged**